submariner/globalnet-ovn.md

```diff
@@ -0,0 +1,122 @@
+   * The NPS runs as a K8s deployment with a replica count of 1.
+      It is primarily used to interact with the OVN-Kubernetes CNI and is responsible for creating a networking
+   topology in the OVN to route the remote cluster traffic via the active Gateway node.
+   * It achieves this by creating a logical `submariner_router`, `submariner_gateway` and `submariner_join` switches
+   in the OVN pipeline and associates the OVN-K's main Load Balancer Group (named clusterLBGroup) to the
+   `submariner_router` to provide service connectivity between the clusters.
+   * When a remote endpoint notification is received by the NPS controller, it programs Logical Router Policies in
+   `ovn_cluster_router` to redirect the remote cluster traffic via the logical `submariner_router`.
+   * Similarly, when a local endpoint notification is received by the NPS controller, it looks at the hostname info,
+   pins the `submariner_router` to the chassis of the active Gateway node and programs Logical Routes on the
+   `submariner_router` to redirect incoming traffic from the remote-cluster to the `ovn_cluster_router` on the Gateway
+   node.
+2. OVN Handler in Route Agent:
+   * It creates the `br-submariner` bridge with an internal interface `ovn-k8s-sub0` configured with `169.254.254.10` IP address. This is done on all the worker nodes, but it's mainly used on the Gateway node alone.
+   * Similarly, it creates `SUBMARINER-POSTROUTING` chain in the `nat` table and `SUBMARINER-FWD-MSSCLAMP`/`SUBMARINER-FORWARD` chains in the `filter` table.
+   * On the active Gateway node of the cluster, these chains are programmed with necessary IPtable rules to allow the
+   inter-cluster traffic.
+   * When a remote endpoint notification is received by the handler, it programs routes in `table 150` on all the
+   worker nodes with default next hop present on the `ovn-k8s-mp0` interface.
+   Basically, this is done to support pods with hostNetworking to connect to a remote service.
+   The idea here is to send the traffic via the CNI network and not via the default interface on the host.
+   * Similarly, on the Gateway node, it programs
+     * routes in `table 149` to redirect the incoming traffic from the remote cluster via the `submariner_router` and 
+     * IPtable rules in `SUBMARINER-POSTROUTING` chain of `nat` table not to perform any NAT operation on this traffic.
+
+![SubmarinerOVNTopology](./images/ovn-submariner-architecture.svg)
+
+So, in the Submariner OVN architecture the inter-cluster traffic touches the host before leaving the Gateway node via
+the tunnel. Because of this, we can leverage the current Globalnet design which programs the iptable rules on the
+Gateway node even for OVN-Kubernetes deployments. 
+
+Naturally, a question would arise if this architecture would work going forward. 
+The answer would be yes because of the way we are creating the Submariner pipeline in OVN.
```

Comment:
This isn’t obvious to me — what potential problems give rise to the question? How does our method of creating the pipeline address them?

Reply:
Please take a look at the updated description.